### PR TITLE
[Feature] Link to mobile settings page for permissions

### PR DESCRIPTION
**Pull request by Leah-Xia-Microsoft**
## Purpose

* Add Go To Settings button to access OS rights via shortcut access

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* Check Go To Settings button can lead to OS access rights.

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/Setup/SetupViewComponent/PreviewAreaViewModel.swift [hunk 1]
```diff
@@ -20,6 +22,20 @@ class PreviewAreaViewModel: ObservableObject {
...
         localVideoViewModel = compositeViewModelFactory.makeLocalVideoViewModel(dispatchAction: dispatchAction)
         self.localizationProvider = localizationProvider
+
+        goToSettingsButtonViewModel = compositeViewModelFactory.makePrimaryButtonViewModel(
+            buttonStyle: .primaryOutline,
+            buttonLabel: self.localizationProvider
+                .getLocalizedString(.goToSettings),
+            iconName: .none,
+            isDisabled: false) { [weak self] in
+                guard let self = self else {
```
Comment (jpeng-ms) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/Setup/SetupViewComponent/PreviewAreaViewModel.swift:32]: maybe use unowned self to get rid of boilerplate code from 32-34?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/Setup/SetupViewComponent/PreviewAreaViewModel.swift [hunk 2]
```diff
@@ -61,4 +77,12 @@ class PreviewAreaViewModel: ObservableObject {
             self.isPermissionsDenied = isPermissionDenied
         }
     }
+
+    private func goToSettingsButtonTapped() {
+        if let url = URL(string: UIApplication.openSettingsURLString) {
```
Comment (jpeng-ms) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Presentation/SwiftUI/Setup/SetupViewComponent/PreviewAreaViewModel.swift:82]: how about using guard to early exit?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Localization/en.lproj/Localizable.strings [hunk 3]
```diff
@@ -26,6 +26,7 @@
 "AzureCommunicationUICalling.SetupView.Button.MicOn.AccessibilityLabel"="Mute";
 "AzureCommunicationUICalling.SetupView.Button.Device"="Device";
 "AzureCommunicationUICalling.SetupView.Button.Device.AccessibilityLabel"="Select audio device";
+"AzureCommunicationUICalling.SetupView.Button.GoToSettings"="Go to Settings";
```
Comment (jpeng-ms) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Localization/en.lproj/Localizable.strings:29]: hi I might be wrong, but I remember Jim told me that we only need one locale here and others would be generated when it's merged to develop. so I am not sure if we need en-GB and en-US. But let me know if I am wrong

Comment (Leah-Xia-Microsoft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Localization/en.lproj/Localizable.strings:29]: > hi I might be wrong, but I remember Jim told me that we only need one locale here and others would be generated when it's merged to develop. so I am not sure if we need en-GB and en-US. But let me know if I am wrong

I'll remove those. Thanks.

**Review by emlynmac — DISMISSED** [review]
LGTM

**Review by jpeng-ms — APPROVED** [review]
Looks good!

**Review by jimchou-dev — APPROVED** [review]
Tested, LGTM

**Pull request merged**